pkg/ui/react-app/src/thanos/pages/blocks/helpers.ts

```diff
@@ -10,14 +10,19 @@ const stringify = (map: LabelSet): string => {
 
 const sortBlocksInRows = (blocks: Block[]): BlocksPool => {
   const pool: BlocksPool = {};
+  const map = new Map(); //map to keep track of similar blocks.
 
   blocks
     .sort((a, b) => a.thanos.downsample.resolution - b.thanos.downsample.resolution)
     .forEach(b => {
-      if (!pool[`${b.compaction.level}-${b.thanos.downsample.resolution}`])
-        pool[`${b.compaction.level}-${b.thanos.downsample.resolution}`] = [];
+      if (!map.has(`${b.maxTime}-${b.minTime}`)) map.set(`${b.maxTime}-${b.minTime}`, 0);
+      else map.set(`${b.maxTime}-${b.minTime}`, map.get(`${b.maxTime}-${b.minTime}`) + 1);
+      const count = map.get(`${b.maxTime}-${b.minTime}`);
 
-      pool[`${b.compaction.level}-${b.thanos.downsample.resolution}`].push(b);
+      if (!pool[`${b.compaction.level}-${b.thanos.downsample.resolution}-${count}`])
+        pool[`${b.compaction.level}-${b.thanos.downsample.resolution}-${count}`] = [];
+
+      pool[`${b.compaction.level}-${b.thanos.downsample.resolution}-${count}`].push(b);
```

Comment:
This will only work when we have perfectly overlapping blocks, i.e. when both the `minTime` and `maxTime` is exactly same for these blocks. But we can have partially overlapping blocks as well. The condition we need to check is if a block's `minTime` is less than the `maxTime` of the previous block.